pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -1158,52 +1250,300 @@ int kretprobe__sys_socket(struct pt_regs* ctx) {
 
 //endregion
 
-// This function is meant to be used as a BPF_PROG_TYPE_SOCKET_FILTER.
-// When attached to a RAW_SOCKET, this code filters out everything but DNS traffic.
-// All structs referenced here are kernel independent as they simply map protocol headers (Ethernet, IP and UDP).
-SEC("socket/dns_filter")
-int socket__dns_filter(struct __sk_buff* skb) {
+static __always_inline __u64 read_conn_tuple_skb(struct __sk_buff* skb, skb_info_t* info) {
+    __builtin_memset(info, 0, sizeof(skb_info_t));
+    info->data_off = ETH_HLEN;
+
     __u16 l3_proto = load_half(skb, offsetof(struct ethhdr, h_proto));
     __u8 l4_proto;
-    size_t ip_hdr_size;
-    size_t src_port_offset;
-    size_t dst_port_offset;
-
     switch (l3_proto) {
     case ETH_P_IP:
-        ip_hdr_size = sizeof(struct iphdr);
-        l4_proto = load_byte(skb, ETH_HLEN + offsetof(struct iphdr, protocol));
-        break;
-    case ETH_P_IPV6:
-        ip_hdr_size = sizeof(struct ipv6hdr);
-        l4_proto = load_byte(skb, ETH_HLEN + offsetof(struct ipv6hdr, nexthdr));
+        l4_proto = load_byte(skb, info->data_off + offsetof(struct iphdr, protocol));
+
+        info->tup.metadata |= CONN_V4;
+        info->tup.saddr_l = load_word(skb, info->data_off + offsetof(struct iphdr, saddr));
+        info->tup.daddr_l = load_word(skb, info->data_off + offsetof(struct iphdr, daddr));
+        info->data_off += sizeof(struct iphdr); // TODO: this assumes there are no IP options
         break;
     default:
+        // TODO: add case for ETH_P_IPV6
```

Comment:
You still have a TODO here for IPv6

Reply:
I think it's fine to merge it without IPv6 support for now

Reply:
This would be a regression because DNS supports IPv6 right now.

Reply:
Oh, that's a very good point. I forgot this code path is shared 😳

Reply:
I opened a PR with an integration test for DNS on IPv6 to make sure there won't be any regressions: https://github.com/DataDog/datadog-agent/pull/6944